demos/minimal/actors/sample_generator.py

```diff
@@ -1,68 +1,79 @@
 from improv.actor import Actor, RunManager
-from datetime import date  # used for saving
 import numpy as np
 import logging
+import time  # Importing time module for the delay
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
 class Generator(Actor):
-    """Sample actor to generate data to pass into a sample processor.
-
+    """Sample actor generate sine/cosine waves based on odd/even frame numbers respectively to pass into a sample processor.
     Intended for use along with sample_processor.py.
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.data = None
         self.name = "Generator"
-        self.frame_num = 0
+        self.frame_num = 0  # Initialize frame counter
+        self.data = None
+        self.max_frames = 20  # Set the limit for number of frames
 
     def __str__(self):
         return f"Name: {self.name}, Data: {self.data}"
 
     def setup(self):
-        """Generates an array that serves as an initial source of data.
-
-        Initial array is a 100 row, 5 column numpy matrix that contains
-        integers from 1-99, inclusive.
-        """
-
+        """Initial setup for Generator"""
```

Comment:
I would add an additional comment below this line (similar to how it was before) that says what the initial array is.